capstone/capapi/routers.py

```diff
@@ -0,0 +1,44 @@
+class CAPAPIRouter(object):
+    """
+    Determine how to route database calls for an app's models (in this case, for an app named capapi).
+    All other models will be routed to the next router in the DATABASE_ROUTERS setting if applicable,
+    or otherwise to the default database.
+    """
+
+    def db_for_read(self, model, **hints):
+        """Send all read operations on capapi app models to `capapi`."""
+        if model._meta.app_label == 'capapi':
+            return 'capapi'
+        return None
+
+    def db_for_write(self, model, **hints):
+        """Send all write operations on capapi app models to `capapi`."""
+        if model._meta.app_label == 'capapi':
+            return 'capapi'
+        return None
+
+    def allow_relation(self, obj1, obj2, **hints):
+        """Determine if relationship is allowed between two objects."""
+        # Allow any relation between two models that are both in the capapi app.
+        if obj1._meta.app_label == 'capapi' and obj2._meta.app_label == 'capapi':
+            return True
+        # No opinion if neither object is in the capapi app (defer to default or other routers).
+        elif 'capapi' not in [obj1._meta.app_label, obj2._meta.app_label]:
+            return None
+
+        # Block relationship if one object is in the capapi app and the other isn't.
+            return False
+
+    def allow_migrate(self, db, app_label, model_name=None, **hints):
+        """Ensure that the capapi app's models get created on the right database."""
+        if db == 'capapi' and app_label == 'capapi':
+            return True
+        if app_label == 'capapi':
```

Comment:
This last if-elif is a little confusing as a way to express this test -- I think what it boils down to is return false if db or app_label is capapi, which is the same test you're running up in `allow_relation` but expressed in a very different way.

Suggestion: this router ultimately implements exactly the same logic as `tracking_tool.routers.TrackingToolDatabaseRouter`, except with the string 'capapi' instead of 'tracking_tool'. Rather than have two redundant routers, could we create a `class SeparateDatabaseRouter` that implements the TrackingToolDatabaseRouter logic with `self.database_name` in place of `'tracking_tool'`, and subclass it as `class CAPAPIRouter(SeparateDatabaseRouter): database_name = 'capapi'`, `class TrackingToolDatabaseRouter(SeparateDatabaseRouter): database_name = 'tracking_tool'`?